bug(frontend): TypeScript build fails — ForceGraph and SearchPage type errors

## Bug Report

**Found in:** `make infra` → `docker-compose build frontend` → `tsc && vite build`

### Description

The frontend TypeScript compilation fails with 3 errors across 2 files:

#### 1. `ForceGraph.tsx:37` — missing `community_id` on `InternalNode`
```
error TS2741: Property 'community_id' is missing in type '{ id: string; label: string; 
name_ar: string; name_en: string | null; type: string; generation: string | null; }' 
but required in type 'InternalNode'.
```

#### 2. `ForceGraph.tsx:98` — `nodeCanvasObject` callback type mismatch
```
error TS2322: Type '(node: InternalNode, ctx: CanvasRenderingContext2D) => void' is not 
assignable to type 'CanvasCustomRenderFn<...>'.
Types of parameters 'node' and 'obj' are incompatible.
Type '...' is missing the following properties from type 'InternalNode': label, community_id, type
```

#### 3. `SearchPage.tsx:17` — missing required argument
```
error TS2554: Expected 1 arguments, but got 0.
```

### Steps to Reproduce

1. Run `make infra`
2. Frontend container build fails at `RUN npm run build` (`tsc && vite build`)

### Root Cause

- `InternalNode` type requires `community_id` but node objects are constructed without it
- The `nodeCanvasObject` callback uses `InternalNode` which doesn't match the library's generic type
- A function in `SearchPage.tsx` is called without its required argument

### Acceptance Criteria

- [ ] `InternalNode` type and all node construction sites are consistent (include `community_id`)
- [ ] `nodeCanvasObject` callback type is compatible with `react-force-graph-2d`
- [ ] `SearchPage.tsx` function call includes required argument
- [ ] `npm run build` (`tsc && vite build`) completes successfully
- [ ] `make infra` builds the frontend container without errors

--- a/frontend/src/components/ForceGraph.tsx
+++ b/frontend/src/components/ForceGraph.tsx
@@ -8,7 +8,7 @@ const COMMUNITY_COLORS = [
   '#aec7e8', '#ffbb78', '#98df8a', '#ff9896', '#c5b0d5',
 ]
 
-function communityColor(id: number | null): string {
+function communityColor(id: number | null | undefined): string {
   if (id == null) return '#999'
   return COMMUNITY_COLORS[id % COMMUNITY_COLORS.length] ?? '#999'
 }
@@ -24,10 +24,11 @@ interface ForceGraphProps {
 interface InternalNode {
   id: string
   label: string
-  community_id: number | null
+  community_id?: number | null
   type: string
   x?: number
   y?: number
+  [key: string]: unknown
 }
 
 export default function ForceGraph({ nodes, edges, onNodeClick, width, height }: ForceGraphProps) {
@@ -63,7 +64,8 @@ export default function ForceGraph({ nodes, edges, onNodeClick, width, height }:
   )
 
   const nodeCanvasObject = useCallback(
-    (node: InternalNode, ctx: CanvasRenderingContext2D) => {
+    (obj: object, ctx: CanvasRenderingContext2D) => {
+      const node = obj as InternalNode
       const x = node.x ?? 0
       const y = node.y ?? 0
       const radius = 6

--- a/frontend/src/pages/SearchPage.tsx
+++ b/frontend/src/pages/SearchPage.tsx
@@ -14,7 +14,7 @@ export default function SearchPage() {
   const [query, setQuery] = useState('')
   const [mode, setMode] = useState<'fulltext' | 'semantic'>('fulltext')
   const navigate = useNavigate()
-  const debounceRef = useRef<ReturnType<typeof setTimeout>>()
+  const debounceRef = useRef<ReturnType<typeof setTimeout> | undefined>(undefined)
 
   useEffect(() => {
     if (debounceRef.current) clearTimeout(debounceRef.current)

